### DISCUSSION TITLE
Android build of Flutter app failing today

### DISCUSSION THREAD

**Discussion by 4bSolutionsLLP** [codemagic-ci-cd/codemagic-docs#2613] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

Normal codemagic build that has previously worked

### Expected results

Expected to build android release without errors

### Actual results

Whilst iOS build is fine, the android build shows the following error, even on build that worked fine previously. 

Execution failed for task ':app:processReleaseMainManifest'.
> Unable to make field private final java.lang.String java.io.File.path accessible: module java.base does not "opens java.io" to unnamed module @3e6fefd

**Comment by dam-ease** [comment]
Hello @4bSolutionsLLP,

Can you please share the buildID for the build that failed?
You can find it under your build details page's 'ID' section.
﻿Navigate to the 'Builds' section on the left-hand menu, select the build you want to view, and click the copy icon.

Also, from the error you shared, please check out https://github.com/fluttercommunity/flutter_workmanager/issues/287#issuecomment-893476072 and this SO thread https://stackoverflow.com/questions/67782975/how-to-fix-the-module-java-base-does-not-opens-java-io-to-unnamed-module to see if the solutions work in your case.

**Comment by 4bSolutionsLLP** [comment]
Thanks. The build ID id 667d29a95f11e7e613268039.

It looks like you have changed something your end as previous Android builds that worked, now don't.

Regards
Ian


On 27/06/2024 09:50:37, Damilola Alimi ***@***.***> wrote:
Hello @4bSolutionsLLP [https://github.com/4bSolutionsLLP],
Can you please share the buildID for the build that failed?
You can find it under your build details page's 'ID' section.
﻿Navigate to the 'Builds' section on the left-hand menu, select the build you want to view, and click the copy icon.
Also, from the error you shared, please check out fluttercommunity/flutter_workmanager#287 (comment) [https://github.com/fluttercommunity/flutter_workmanager/issues/287#issuecomment-893476072] and this SO thread https://stackoverflow.com/questions/67782975/how-to-fix-the-module-java-base-does-not-opens-java-io-to-unnamed-module [https://stackoverflow.com/questions/67782975/how-to-fix-the-module-java-base-does-not-opens-java-io-to-unnamed-module] to see if the solutions work in your case.
—
Reply to this email directly, view it on GitHub [https://github.com/orgs/codemagic-ci-cd/discussions/2613#discussioncomment-9891517], or unsubscribe [https://github.com/notifications/unsubscribe-auth/AQDXTEWRF752KXOTGHN2RK3ZJPG5XAVCNFSM6AAAAABJ5MAH66VHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4TQOJRGUYTO].
You are receiving this because you were mentioned.Message ID: ***@***.***>

--
This email has been checked for viruses by AVG antivirus software.
www.avg.com

**Reply by dam-ease** [reply, marked as answer]
Thanks for your response!
I see you are on an older version of Flutter and this seems like a gradle issue. You can try using APG version com.android.tools.build:gradle:7.3.1 to see if it works or the workarounds in https://github.com/fluttercommunity/flutter_workmanager/issues/287#issuecomment-893476072 and this SO thread https://stackoverflow.com/questions/67782975/how-to-fix-the-module-java-base-does-not-opens-java-io-to-unnamed-module

**Comment by 4bSolutionsLLP** [comment]
Many thanks - will give it a try. On Fri, 28 Jun 2024 at 15:35, Damilola Alimi ***@***.***> wrote:
Thanks for your response!
I see you are on an older version of Flutter and this seems like a gradle issue. You can try using APG version com.android.tools.build:gradle:7.3.1 to see if it works or the workarounds in fluttercommunity/flutter_workmanager#287 (comment) and this SO thread https://stackoverflow.com/questions/67782975/how-to-fix-the-module-java-base-does-not-opens-java-io-to-unnamed-module

**Comment by youngminz** [comment]
I can confirm that the AGP version upgrade worked! I've upgraded from AGP 4.2.2 to 7.4.2 using the `Tools -> AGP Upgrade Assistant`, and the error has gone. Many thanks to @dam-ease, you saved my day.

**Reply by dam-ease** [reply]
Glad it did!
Thank you for the feedback.
